moto/ec2/models.py

```diff
@@ -1810,6 +1811,10 @@ def get_volume(self, volume_id):
 
     def delete_volume(self, volume_id):
         if volume_id in self.volumes:
+            volume = self.volumes[volume_id]
+            instance_id = volume.attachment.instance.id
+            if volume.attachment is not None:
```

Comment:
This method will always raise an error because if `volume.attachment` _is_ `None` then `volume.attachment.instance` would have raised an error a line before. Maybe you want to move line 1815 inside this conditional?